mne/stats/cluster_level.py

```diff
@@ -373,12 +374,14 @@ def _find_clusters(
     show_info : bool
         If True, display information about thresholds used (for TFCE). Should
         only be done for the standard permutation.
+    clusters_out : bool
+        If True, clusters are returned, otherwise None is returned instead
```

Comment:
Do we need this param at all? Could we instead just effectively have `clusters_out = not tfce`?

Reply:
Good observation. Indeed the clusters_out point is to not make the long list of arrays at each call of _find_cluster during permutation computation.
Since 'clusters' with TFCE is simply a list of one cluster per point, we could just decide to make this construction (for TFCE only) outside of the _find_cluster function. This would still need to depend on the dimension of input (1d or 2d) and the type of output wanted 'indices' or 'mask'. 
I do not know what is the simplest (or more compliant with MNE practices) between the current additional parameter and moving the TFCE specific lines (lines 494-505) in _permutation_cluster_test (somewhere around lines1021 to 1033). But yes I would agree with the moving.
Any advice ?

Reply:
+1 for just doing the construction outside of the `_find_cluster` function wherever it's needed. As long as the public API output stays the same (and hopefully this is checked in a test already, if not then please add it!) then we should be okay to refactor however is cleanest

Reply:
Done, I moved the "clusters" construction for TFCE out of `_find_clusters`.

Regarding the API, there is already a test `test_output_equiv` to which I added the `threshold` parameter to be tested. 
However, I also noticed the `False` value for `adjacency` was not tested and indeed there are problems here:
- `adjacency=False` runs only for 1D inputs (which I think is not expected)
- for 1D input, the output is always `"indices"` whatever is `out_type`

I guess it is rarely used... Should this be corrected in the same or another PR?

Reply:
> I guess it is rarely used... Should this be corrected in the same or another PR?

Let's do it in another PR, I created https://github.com/mne-tools/mne-python/issues/12613 so we don't forget